File: kotlin/scrabble-score/src/main/kotlin/ScrabbleScore.kt
object ScrabbleScore {
    val letterToPointValue = mapOf(
            "AEIOULNRST" to 1,
            "DG"    to 2,
            "BCMP"  to 3,
            "FHVWY" to 4,
            "K"     to 5,
            "JX"    to 8,
            "QZ"    to 10
    )

    fun scoreLetter(c: Char): Int {
        val upperCasedCharacter = c.uppercase()

        for (key in letterToPointValue.keys) {
            if (key.contains(upperCasedCharacter)) {
                return letterToPointValue[key] ?: 0
            }
        }

        return 0;
    }

    fun scoreWord(word: String): Int {
        return  word.map { c -> scoreLetter(c) }.sum();
    }
}

File: python/scrabble-score/scrabble_score.py
_score_by_letter = (
    ("aeioulnrst", 1),
    ("dg", 2),
    ("bcmp", 3),
    ("fhvwy", 4),
    ("k", 5),
    ("jx", 8),
    ("qz", 10)
)

SCORES = {}
for letter, score in _score_by_letter:
    SCORES.update((dict.fromkeys(letter, score)))


def score(word):
    return sum(SCORES[char] for char in word.lower())
